mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs

```diff
@@ -372,14 +375,16 @@ public ResourceManager(String baseName, Assembly assembly)
 
             CommonAssemblyInit();
 
-            m_callingAssembly = (RuntimeAssembly)Assembly.GetCallingAssembly();
-            // Special case for mscorlib - protect mscorlib's private resources.
-            // This isn't for security reasons, but to ensure we can make
-            // breaking changes to mscorlib's internal resources without 
-            // assuming users may have taken a dependency on them.
-            if (assembly == typeof(Object).Assembly && m_callingAssembly != assembly)
-            {
-                m_callingAssembly = null;
+            // GetCallingAssembly might not work on some platforms
+            try {
+                m_callingAssembly = (RuntimeAssembly)Assembly.GetCallingAssembly();
+                // Special case for mscorlib - protect mscorlib's private resources.
+                // This isn't for security reasons, but to ensure we can make
+                // breaking changes to mscorlib's internal resources without 
+                // assuming users may have taken a dependency on them.
+                if (assembly == typeof(Object).Assembly && m_callingAssembly != assembly)
+                    m_callingAssembly = null;
+            } catch {
```

Comment:
Do we want the empty catch to apply to all profiles?  I don't have a sense of the implication.